app/views/layouts/_index.html.erb

```diff
@@ -0,0 +1,69 @@
+<div class="row">
+  <div class="col-md-12">
+    <div class="panel panel-default">
+      <div class="panel-body">
+        <h1 class="index-header text-center">
+          <%= @page_title %>
+        </h1>
+      </div>
+    </div>
+  </div>
+</div>
+
+<% if @new_button %>
+  <div class="row col-md-12 index-new-button">
+    <a href=<%= @new_button[:url] %> role="button" class="btn btn-success"> <%= @new_button[:text] %> </a>
+  </div>
+<% end %>
+
+<table id="indexTable" class="display hover order-column compact" width="100%">
+  <thead>
+    <tr>
+      <% @headers.each do |header| %>
+        <th> <%= header %> </th>
+      <% end %>
+  </thead>
+
+  <% @models.each do |model| %>
+    <% if @clickable_rows %>
+      <tr class="clickable" data-url=<%= url_for(model) %>>
+    <% else %>
+      <tr>
+    <% end %>
+    <% @columns.each do |column| %>
+      <td> <%= model.send("#{column}") %> </td>
```

Comment:
It's a dynamic way to call a function. You can send a string and if a function with that name exists on the model it will call it. We need to call these as functions instead of attribute accessors because some of these (specifically `full_address`) are methods on the model instead of actual columns. ):